### PR TITLE
Refactor tunnel issue in UI

### LINKED ISSUE #2507: Remove the security report's dependency on the `Monkey` model
# Refactor

## Component(s) to be refactored

- `monkey_island.cc.services.reporting.ReportService`

## Explanation

The `Monkey` model has been superseded by the `Agent` model. Various parts of the `Monkey` model have been broken out into other components, such as events or signals. Decouple the `ReportService` from the `Monkey` model.

## Tasks

- [x] Understand the slowest part of report generation and how it scales (0d) @ilija-lazoroski 
- [x] Either remove caching or cache based on latest event (0d) @ilija-lazoroski 
- [x] Remove `aws_instance_id` from report generation (0d) - @shreyamalviya 
- [ ] Refactor the "tunnel issue" report component in the UI (0.5d) - @shreyamalviya 
    - [x] Pull machines and node communications and determine if any tunneling was used
    - [x] Display the "tunnel issues"
    - [x] Remove "tunnel issue" generation from the backend
- [x] Refactor the UI to use the logic from #2509 to get "manual monkeys" (0d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #2507

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the Island and checking the report with dummy data
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 1]
```diff
@@ -163,6 +170,16 @@ class ReportPageComponent extends AuthComponent {
       })
   }
 
+  getAllAgents() {
+    IslandHttpClient.get(APIEndpoint.agents)
+      .then(res => this.setState({agents: res.body}))
+  }
+
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:177]: We already have a function for this in `ServerUtils`.

**Comment by shreyamalviya** [comment]
Updated screenshot:
image

**Review by ilija-lazoroski — APPROVED** [review]
LGTM!

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 2]
```diff
@@ -227,7 +232,7 @@ class ReportPageComponent extends AuthComponent {
   }
 
   generateReportOverviewSection() {
-    let manual_monkey_hostnames = getManuallyStartedAgents(this.state.agents).map((agent) => getMachineHostname(getAgentMachine(agent, this.state.machines)));
+    let manual_monkey_hostnames = getManuallyStartedAgents(this.state.agents).map((agent) => getMachineHostname(getMachineByAgent(agent, this.state.machines)));
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:235]: Switch to camelCase

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 3]
```diff
@@ -526,6 +537,45 @@ class ReportPageComponent extends AuthComponent {
     return <ul>{issuesDivArray}</ul>;
   };
 
+  createTunnelingIssueComponent(agents, machines) {
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:540]: Can we move this to the `TunnelIssue.js`?

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 4]
```diff
@@ -526,6 +537,45 @@ class ReportPageComponent extends AuthComponent {
...
+            agentTunnelMachineHostname = getMachineHostname(agentTunnelMachine);
+          }
+
+          tunnelingIssues.push({
+            'agent_machine': agentMachineHostname,
+            'agent_tunnel': agentTunnelMachineHostname
+          });
+        }
+      }
+    }
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:571]: This logic looks like it belongs in the `tunnelIssueReport.js` component

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 5]
```diff
@@ -356,6 +361,11 @@ class ReportPageComponent extends AuthComponent {
         overviews.push(this.getIssueOverview(this.IssueDescriptorEnum[issues[i]]));
       }
     }
+
+    if (this.tunnelingIssueExists === true) {
+      overviews.push(this.getIssueOverview(this.IssueDescriptorEnum['tunnel']))
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:366]: Always push the tunneling issue overview. The overview will return `<></>` if no tunneling happened

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 6]
```diff
@@ -444,6 +454,7 @@ class ReportPageComponent extends AuthComponent {
           <h3>Machine related recommendations</h3> : null}
         <div>
           {this.generateIssues(this.state.report.recommendations.issues)}
+          {this.tunnelingIssueComponent}
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:457]: We don't need to store this in the state. We can just call `tunnelIssueReport` component here and pass machines and agents. If there's no tunneling issue, the component can return `<></>`

File: monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx [hunk 7]
```diff
@@ -54,6 +61,20 @@ export function getMachineHostname(machine): string {
   return hostname;
 }
 
+export function getMachineFromIP(ip, machines) {
+  let machineFromIP = null;
+
+  for (let machine of machines) {
+    let machineIPs = machine['network_interfaces'].map(network_interface => network_interface.split('/')[0]);
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx:68]: We're duplicating This logic all over the place. See also https://github.com/guardicore/monkey/pull/2553. Let's make his reusable.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js [hunk 8]
```diff
@@ -109,7 +115,6 @@ class ReportPageComponent extends AuthComponent {
       },
       'tunnel': {
         [this.issueContentTypes.OVERVIEW]: tunnelIssueOverview,
-        [this.issueContentTypes.REPORT]: tunnelIssueReport,
         [this.issueContentTypes.TYPE]: this.issueTypes.WARNING
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js:118]: Remove this whole thing, if we turn overview and report into standalone components, they can determine by machines and agents if they should render or not

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Move the logic to tunneling issue if possible

File: monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx (hunk 7 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx:68]: Done in #2553.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**